### PR TITLE
upcoming: [M3-7481] - Disable creating and editing API tokens for proxy users

### PR DESCRIPTION
**Pull request by mjac0bs**
## Description 📝
According to the API spec:

Proxy users may not generate additional tokens for themselves.
Proxy users will be able to view the tokens provisioned for them, but will not be able to modify them.
Proxy users should still be able to revoke their own PAT.

## Changes  🔄

- The "Create A Personal Access Token" button is disabled for proxy users.
- The disabled button has a tooltip that reads: "You can only create API tokens for your own company."
- Proxy users can view the tokens provisioned for them but cannot modify them.
- The disabled Rename action has a tooltip that reads: "Only company users can edit API tokens."
- Added Cypress test coverage for proxy user disabled PAT states.

## Preview 📷

| Description  | Screenshot   |
| ------- | ------- |
| Disabled Edit action button |Screenshot 2024-01-24 at 3 40 29 PM |
| Disabled Create button |Screenshot 2024-01-24 at 3 40 09 PM |

### Prerequisites
(How to setup test environment)
- Check out this PR.
- Make sure the Parent/Child feature flag is on and change `user_type` to `proxy` in [`serverHandlers.ts`](https://github.com/linode/manager/blob/develop/packages/manager/src/mocks/serverHandlers.ts#L554).

### Verification steps 
(How to verify changes)
- As a proxy user, go to http://localhost:3000/profile/tokens.
- Confirm that the `Create a Personal Access Token` is disabled and hovering over the button displays a tooltip explaining the reason for disabling.
- Confirm that the edit action (`Rename`) is disabled and hovering over the help tooltip displays a message explaining the reason for disabling.
- Confirm that API tokens are editable and creatable for non proxy users (e.g. other user_types; feature flag off).
- Confirm test passes:
```
yarn cy:run -s "cypress/e2e/core/account/personal-access-tokens.spec.ts"
```

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

**Review by bnussman-akamai — APPROVED** [review]
All the verification steps looked good. Thanks for including the test!

File: packages/manager/cypress/e2e/core/account/personal-access-tokens.spec.ts [hunk 1]
```diff
@@ -224,4 +231,107 @@ describe('Personal access tokens', () => {
...
+      .should('be.disabled')
+      .click();
+
+    cy.findByRole('tooltip')
+      .should('be.visible')
+      .within(() => {
+        cy.findByText(
+          'You can only create tokens for your own company.'
+        ).should('be.visible');
+      });
```
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/account/personal-access-tokens.spec.ts:281]:
```suggestion
    ui.tooltip
      .findByText('You can only create tokens for your own company.')
      .should('be.visible');
```
Just pointing out that this exists for the future!

The helper is a little bit more ergonomic, and has the extra advantage of working from inside `within(() => {})` callbacks even though the tooltip itself is often somewhere else in the DOM (not applicable to your situation of course)

**Review by jdamore-linode — APPROVED** [review]
This is great, thanks for the tests @mjac0bs! This made me realize I missed this flow when I wrote up the test tickets for this project, I appreciate you covering it here!

File: packages/manager/cypress/e2e/core/account/personal-access-tokens.spec.ts (hunk 1 above)
Comment (mjac0bs) [reply packages/manager/cypress/e2e/core/account/personal-access-tokens.spec.ts]: Nice, thanks! I appreciate the reminder that these UI helpers exist; made that quick change in [3a35491](https://github.com/linode/manager/pull/10109/commits/3a35491e60b64327a3eec23d6c3f2807568adf6e).

**Pull request merged**